### PR TITLE
add tyk_context to response body transform

### PR DESCRIPTION
**Pull request by asoorm**
When `EnableContextVars: true`, request context data is added to
`bodyData["_tyk_context"]` for use within response body transform
templates.
    
Similarly, we have also included `bodyData["_tyk_meta"]`.

### CODE REVIEW

File: api_loader.go [hunk 1]
```diff
@@ -406,8 +406,8 @@ func processSpec(spec *APISpec, apisByListen map[string]int,
 		mwAppendEnabled(&chainArray, &StripAuth{baseMid})
 		mwAppendEnabled(&chainArray, &KeyExpired{baseMid})
 		mwAppendEnabled(&chainArray, &AccessRightsCheck{baseMid})
-		mwAppendEnabled(&chainArray, &RateLimitForAPI{BaseMiddleware: baseMid})
 		mwAppendEnabled(&chainArray, &RateLimitAndQuotaCheck{baseMid})
+		mwAppendEnabled(&chainArray, &RateLimitForAPI{BaseMiddleware: baseMid})
```
Comment (letzya) [inline api_loader.go]: Is this related? or just testing?

Comment (asoorm) [reply api_loader.go]: not related - was a checkout of another branch

File: res_handler_transform.go [hunk 2]
```diff
@@ -103,11 +103,23 @@ func (h *ResponseTransformMiddleware) HandleResponse(rw http.ResponseWriter, res
...
+		case []interface{}:
+			logger.Debug("got array: []interface{}")
+			bodyData["array"] = tempBody
+		case map[string]interface{}:
+			logger.Debug("got object: map[string]interface{}")
+			bodyData = tempBody.(map[string]interface{})
+		}
 	}
 
+	bodyData["_tyk_context"] = ctxGetData(req)
```
Comment (buger) [inline res_handler_transform.go]: Why only context and not meta?

File: res_handler_transform.go [hunk 3]
```diff
@@ -103,11 +103,23 @@ func (h *ResponseTransformMiddleware) HandleResponse(rw http.ResponseWriter, res
...
+		var tempBody interface{}
+		if err := json.NewDecoder(respBody).Decode(&tempBody); err != nil {
 			logger.WithError(err).Error("Error unmarshalling JSON")
 		}
+
+		switch tempBody.(type) {
+		case []interface{}:
+			logger.Debug("got array: []interface{}")
+			bodyData["array"] = tempBody
+		case map[string]interface{}:
```
Comment (buger) [inline res_handler_transform.go]: Do not think you can do this. Type checking for `map[string]interface{}` does not work. You should use `reflect.TypeOf(a).Kind() == reflect.Map` and `reflect.TypeOf(a).Kind() == reflect.Array`

File: api_loader.go (hunk 1 above)
Comment (buger) [reply api_loader.go]: Can you put link to related PR, so we can merge it, and you will rebase on master.

File: res_handler_transform.go (hunk 3 above)
Comment (dencoded) [reply res_handler_transform.go]: I think it should be OK as json un-marshalling does `map[string]interface{}` or `[]interface{}` if not strict type provided.

It can also be done using type assertion:
```
if tmpVal, ok := tempBody.([]interface{}); ok {
		fmt.Println("got array: []interface{}", tmpVal)
	} else if tmpVal, ok := tempBody.(map[string]interface{}); ok {
		fmt.Println("got object: map[string]interface{}", tmpVal)
	} else {
		fmt.Println("Got something else", tmpVal)
	}
```

also, I think any of those three have about the same performance
